packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts

```diff
@@ -120,6 +121,38 @@ describe('LibraryCommandletExecutor', () => {
     expect(showInfoStub.notCalled).to.be.true;
   });
 
+  it('should show cancellable in progress view', async () => {
+    // const cancelStub = sb.stub(vscodeStub.window, 'showWarningMessage' as any);
+    const withProgressStub = sb.stub(vscodeStub.window, 'withProgress');
+    const reportStub = stub();
+    const progress: Progress<{
+      message?: string;
+      increment?: number;
+    }> = {
+      report: reportStub
+    };
+    const token = new vscodeStub.CancellationTokenSource().token;
+    token.isCancellationRequested = true;
+    withProgressStub.callsFake((options, task) => {
+      task(progress, token);
+    });
+
+    const cancelledExecutor = new TestExecutor(new MockChannel());
+    cancelledExecutor.cancellable = true;
+
+    await cancelledExecutor.execute({ data: { success: true }, type: 'CONTINUE' });
+
+    expect(withProgressStub.called).to.be.true;
+    expect(withProgressStub.getCall(0).args[0]).to.eql({
+      title: nls.localize('progress_notification_text', 'Test Command'),
+      location: vscodeStub.ProgressLocation.Notification,
+      cancellable: true
+    });
+    // expect(cancelStub.calledOnce).to.be.true;
```

Comment:
Still incomplete, but I realized that some of the tests I was trying to build within baseDeployRetrieve weren't achievable and made more sense to have here anyways. I have the mocked token and I am deliberately trying to invoke cancel from it. However, the `token.onCancellationRequested(()` still doesn't fire here. I feel this is close - any ideas on what is missing?

Reply:
I will give that a shot - thank you

Reply:
@xyc This did the trick. Thank you so much!